File: integration-tests/iac/lambdas/node/kinesis_producer.mjs
import { KinesisClient, PutRecordCommand } from '@aws-sdk/client-kinesis';

const kinesis = new KinesisClient();

export async function handler(event, context) {
    const streamName = process.env.STREAM_NAME;

    const message = {
        message: 'Hello from Kinesis producer!',
        request_id: context.awsRequestId,
    };

    const response = await kinesis.send(new PutRecordCommand({
        StreamName: streamName,
        Data: Buffer.from(JSON.stringify(message)),
        PartitionKey: context.awsRequestId,
    }));

    console.log(`Put record to Kinesis: ${response.SequenceNumber}`);

    return {
        statusCode: 200,
        body: JSON.stringify({
            sequence_number: response.SequenceNumber,
        }),
    };
}

File: integration-tests/iac/lambdas/python/kinesis_producer.py
import json
import os
import boto3

kinesis = boto3.client('kinesis')


def handler(event, context):
    stream_name = os.environ['STREAM_NAME']

    message = {
        'message': 'Hello from Kinesis producer!',
        'request_id': context.aws_request_id,
    }

    response = kinesis.put_record(
        StreamName=stream_name,
        Data=json.dumps(message),
        PartitionKey=context.aws_request_id,
    )

    print(f"Put record to Kinesis: {response['SequenceNumber']}")

    return {
        'statusCode': 200,
        'body': json.dumps({
            'sequence_number': response['SequenceNumber'],
        }),
    }
